metaflow/decorators.py

```diff
@@ -158,13 +158,14 @@ def make_decorator_spec(self):
                     attr_list.append("%s=%s" % (k, str(v)))
                 else:
                     attr_list.append("%s=%s" % (k, json.dumps(v).replace('"', '\\"')))
+
             attrstr = ",".join(attr_list)
             return "%s:%s" % (self.name, attrstr)
         else:
             return self.name
 
     def __str__(self):
-        mode = "decorated" if self.statically_defined else "cli"
+        mode = "decorated" if self.statically_defined else "cli or configuration"
```

Comment:
maybe `undecorated`?

Reply:
I find "undecorated" to be less specific. It doesn't really give much information. What do you think of "injected". We could also have "static" and "dynamic" simply. Wdyt?

Reply:
yep - static and dynamic works well.